modules/moderation.py

```diff
@@ -268,350 +330,152 @@ async def _unmuting(self, ctx, member: discord.Member, *, reason='-No reason spe
+                str(member), # Username
+                doc['type'][-1:], # Tier type
+                datetime.datetime.utcfromtimestamp(doc['timestamp']).strftime('%B %d, %Y'), # Date of warn
+                strikeCount, # How many strikes will replace tier,
+                config.parakarry # Parakarry mention for DM
+            )
 
-            else:
-                db.update_one({'_id': warnPun['_id']}, {'$set': {'expiry': int(renew.timestamp())}})
-                await resp.delete()
-                return await ctx.send(f'{config.greenTick} Warning review complete for {str(member)} ({member.id}). Will be delayed {utils.humanize_duration(renew)}')
+            try:
+                await member.send(explanation)
 
-        except asyncio.TimeoutError:
-            await resp.delete()
-            return await ctx.send(content=f'{config.redTick} Command timed out. Rerun to continue.')
+            except discord.Forbidden:
+                failures += 1
+                continue
 
-    @commands.has_any_role(config.moderator, config.eh)
-    @commands.command(name='note')
-    async def _note(self, ctx, user: discord.User, *, content):
-        if len(content) > 900: return await ctx.send(f'{config.redTick} Note is too long, reduce it by at least {len(content) - 990} characters')
-        await utils.issue_pun(user.id, ctx.author.id, 'note', content, active=False, public=False)
-        if await utils.mod_cmd_invoke_delete(ctx.channel):
-            return await ctx.message.delete()
+            except discord.HTTPException as e:
+                failures += 1
+                logging.error(f'[Warn Migration] Failed to migrate {member.id}, {e}')
+                continue
 
-        return await ctx.send(f'{config.greenTick} Note successfully added to {user} ({user.id})')
+        await ctx.send(f'Completed action. Unable to notify {failures} users')
 
     @_banning.error
     @_unbanning.error
     @_kicking.error
+    @_strike.error
+    @_strike_set.error
     @_muting.error
     @_unmuting.error
     @_warning.error
-    @_warning_clear.error
-    @_warning_setlevel.error
-    @_warning_review.error
+    @_strike.error
```

Comment:
consdier using a cog_command_error handler?  https://discordpy.readthedocs.io/en/latest/ext/commands/api.html?highlight=error#discord.ext.commands.Cog.cog_command_error

Reply:
Potentially out of scope for PR as all cogs currently use this logic